### DISCUSSION TITLE
WaveSpeed_c3 is not a number or infinity" Error in Anura3D

### DISCUSSION THREAD

**Discussion by yang-geo22** [Anura3D/Anura3D_OpenSource#161] [Q&A]
Hello Anura3D Community,

I am currently encountering a challenging issue while conducting a Material Point Method (MPM) simulation using Anura3D, specifically in a scenario based on a poroelastic solid interacting with a Newtonian fluid, as described in Section 20.4 of The Material Point Method for Geotechnical Engineering. Despite closely following the material parameters and setup detailed in the reference, I am consistently facing an error that states "WaveSpeed_c3 is not a number or infinity" at time step 1067 of the simulation.My material settings are as follows:
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/94846090/3338299c-1902-4ce3-b675-97e9fbb20ebf)
I have tried adjusting the time step size and examining the mesh quality, but the issue persists. I am reaching out to the community for insights or suggestions that might help resolve this error. Could this be a matter of numerical stability, or are there specific considerations in Anura3D's MPM implementation that I might be overlooking?

Any advice or guidance from fellow users or the Anura3D team would be immensely appreciated. Thank you in advance for your time and assistance.

**Comment by murphy-john** [comment]
Hello, will you provide additional information about your model? Please follow [our advice](https://github.com/Anura3D/Anura3D_OpenSource/discussions/29#discussion-4274028) about posting issue with your models. Do you have any pictures of what you are trying todo? 

Are you using the double-point formulation? If so, it may be beneficial to switch from MPM-MIXED to MPM-MP integration.

**Reply by jiawei6688** [reply]
Hi, i'm having the same problem, Did you figure out how to fix this?
I encountered this problem while simulating rainfall infiltration using the example in Chapter 13.3 of the tutorial manual. I'd like to reproduce it by changing it from a single-point to a two-point simulation. What could be the problem?

pls help me. thank you
<img width="1256" height="832" alt="image" src="https://github.com/user-attachments/assets/c51d12ef-5bb4-4c0d-8805-f91b20eca13c" />
<img width="389" height="789" alt="image" src="https://github.com/user-attachments/assets/7134763e-4771-406f-bca2-9af86028349c" />
<img width="714" height="593" alt="image" src="https://github.com/user-attachments/assets/690d69f9-bce7-4ab9-9249-7b82e065c498" />
<img width="1141" height="843" alt="image" src="https://github.com/user-attachments/assets/d42fed06-01df-4e1c-abed-13b05bc3da60" />
<img width="1150" height="838" alt="image" src="https://github.com/user-attachments/assets/d0e83a37-85ce-4821-ad5c-510fc3a05c2c" />
